client/src/test/java/org/eclipse/hono/client/impl/AbstractSenderTest.java

```diff
@@ -41,20 +47,31 @@
  * Tests verifying behavior of {@code AbstractSender}.
  *
  */
+@RunWith(VertxUnitRunner.class)
 public class AbstractSenderTest {
 
     private ProtonSender protonSender;
     private ClientConfigProperties config;
     private Context context;
 
+    private static Vertx vertx;
+
+    /**
+     * Sets up vertx.
+     */
+    @BeforeClass
+    public static void setUpVertx() {
+        vertx = Vertx.vertx();
```

Comment:
IMHO we should stick with a mocked Vertx instance and make it invoke timer tasks immediately. That way the test case will always run as fast as possible.

Reply:
Good idea. I've changed the test accordingly.